### PR TITLE
Add Optional required field assertions

### LINKED ISSUE #205: Assert that all dataclass attributes of AccountState are optional [complete for 500 TNBC]
The task is to implement this TODO:
```
# TODO(dmu) CRITICAL: Assert all attributes are optional
```

We need to assert that all dataclass attributes of AccountState are optional, so if later a developer creates new attribute which is non-optional the code will fail-fast.

### PR DESCRIPTION
**Pull request by AnonymousDapper**
This should close #205

### CODE REVIEW

File: thenewboston_node/business_logic/models/base.py [hunk 1]
```diff
@@ -18,13 +20,30 @@ def get_request_to_block_type_map():
...
         CoinTransferSignedChangeRequest: (BlockType.COIN_TRANSFER.value, get_updated_account_states_for_coin_transfer),
         NodeDeclarationSignedChangeRequest:
             (BlockType.NODE_DECLARATION.value, get_updated_account_states_for_node_declaration),
     }
 
 
+NoneType = type(None)
+
+
+def assert_all_optional(cls, *, exempt_fields=None):
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/base.py:34]: please, reuse BaseMixin code. We need something simple like:

```
assert all(AccountState.is_optional_field(field_name) for field_name in AccountState.get_field_names())
```

Actually that's all we need

Comment (AnonymousDapper) [reply thenewboston_node/business_logic/models/base.py:34]: It looks like this was already implemented in ae91c6e5, so unless you literally just want me to put the assertion in the code, there's nothing more for me to do here.

Comment (dmugtasimov) [reply thenewboston_node/business_logic/models/base.py:34]: you are about right. but I did not run the code. I need you to make sure that all tests still pass

**Comment by AnonymousDapper** [comment]
Changes implemented as requested. No test errors, 259 pass, 7 are skipped.

File: thenewboston_node/business_logic/models/account_state.py [hunk 2]
```diff
@@ -74,3 +74,5 @@ def validate_node(self):
 
 
 # TODO(dmu) CRITICAL: Assert all attributes are optional
```
Comment (dmugtasimov) [inline thenewboston_node/business_logic/models/account_state.py:76]: @AnonymousDapper This TODO should be removed, since you have just implemented it

Comment (AnonymousDapper) [reply thenewboston_node/business_logic/models/account_state.py]: Done.

**Comment by dmugtasimov** [comment]
@AnonymousDapper please, also squash commits into 1 commit - we do not need 4 commits for 2 lines change. Please, also take care about "All checks have failed". Do you have precommit installed properly?

**Comment by dmugtasimov** [comment]
@AnonymousDapper once you squash commits into one I can merge.

**Comment by AnonymousDapper** [comment]
@dmugtasimov Apologies for the delay, it's been busy here. Everything should be good to go, 37945ff is the only commit in the tree now.

**Comment by dmugtasimov** [comment]
@AnonymousDapper thank you. You will get paid soon

**Pull request merged**